### ISSUE TITLE
code about blip2-t metric

### ISSUE THREAD

**Issue by x2x5** [Rbrq03/ClassDiffusion#16]
Hi, I feel a bit confused about the blip2-t metric in the paper.

In the paper, it says it use BLIP2 model, which should be the paper: **BLIP-2: Bootstrapping Language-Image Pre-training with Frozen Image Encoders and Large Language Models.**

<img width="1042" height="399" alt="Image" src="https://github.com/user-attachments/assets/1c54490d-ecf6-45a9-9d40-f8d2628f3f3c" />

But in the code, you use the model: Salesforce/blip-itm-large-coco, which I think it is the BLIP1 model, the paper is: **BLIP: Bootstrapping Language-Image Pre-training for Unified Vision-Language Understanding and Generation**

<img width="907" height="425" alt="Image" src="https://github.com/user-attachments/assets/f18f3793-f91d-4a2e-9648-e54f1f7de156" />

So do you actually use BLIP1 model to compute the metric?

Or you actually use BLIP2 model? If so, which is the pretrained weight?

<img width="942" height="746" alt="Image" src="https://github.com/user-attachments/assets/aa1c7971-370a-46cf-87ec-39977661280a" />

thanks~

**Comment by Rbrq03** [comment]
Thanks for your interest!
I will come back to this issue and #15 later as I need to check the details of the raw code in our experiments on a long-unused machine for me.
You can set up your experiment with default settings(blip2/blip-itm is both ok), and I tend to think that the slight difference will not significantly affect the conclusion of the experiment.

**Comment by x2x5** [comment]
> Thanks for your interest! I will come back to this issue and [#15](https://github.com/Rbrq03/ClassDiffusion/issues/15) later as I need to check the details of the raw code in our experiments on a long-unused machine for me. You can set up your experiment with default settings(blip2/blip-itm is both ok), and I tend to think that the slight difference will not significantly affect the conclusion of the experiment.

I see, thanks! 

Yeah, it doesn't matter too much; I just want to reproduce the quantitative metrics from the paper as closely as possible. 

Because I feel my results are far away from the results in the table (I kept the original setting of 2 GPUs and the same hyperparameters)

So I think the problem might be in the metric computing part, that's why I asked these questions~

thanks~

| Subject | CLIP-T | CLIP-I | DINO-I | BLIP2-T |
|---------|--------|--------|--------|--------|
| 平均 | 0.252 | 0.727 | 0.487 | 0.252 |
|论文|0.300|0.828|0.673|0.460|
| poop_emoji | 0.268 | 0.547 | 0.227 | 0.267 |
| duck_toy | 0.268 | 0.633 | 0.338 | 0.274 |
| pink_sunglasses | 0.254 | 0.708 | 0.291 | 0.261 |
| rc_car | 0.264 | 0.633 | 0.321 | 0.265 |
| shiny_sneaker | 0.236 | 0.723 | 0.581 | 0.251 |
| dog5 | 0.259 | 0.734 | 0.553 | 0.230 |
| robot_toy | 0.265 | 0.591 | 0.240 | 0.267 |
| dog7 | 0.259 | 0.785 | 0.521 | 0.254 |
| vase | 0.244 | 0.780 | 0.577 | 0.213 |
| candle | 0.255 | 0.781 | 0.566 | 0.260 |
| dog8 | 0.264 | 0.771 | 0.552 | 0.257 |
| monster_toy | 0.265 | 0.548 | 0.226 | 0.265 |
| clock | 0.251 | 0.792 | 0.448 | 0.270 |
| bear_plushie | 0.245 | 0.713 | 0.574 | 0.260 |
| red_cartoon | 0.272 | 0.687 | 0.524 | 0.277 |
| dog6 | 0.248 | 0.835 | 0.666 | 0.250 |
| backpack | 0.227 | 0.842 | 0.505 | 0.259 |
| fancy_boot | 0.234 | 0.774 | 0.520 | 0.213 |
| backpack_dog | 0.247 | 0.679 | 0.461 | 0.239 |
| cat | 0.234 | 0.849 | 0.724 | 0.230 |
| grey_sloth_plushie | 0.245 | 0.706 | 0.538 | 0.259 |
| dog | 0.264 | 0.798 | 0.656 | 0.227 |
| wolf_plushie | 0.248 | 0.691 | 0.418 | 0.252 |
| dog3 | 0.251 | 0.719 | 0.443 | 0.241 |
| dog2 | 0.253 | 0.794 | 0.652 | 0.240 |
| can | 0.249 | 0.680 | 0.527 | 0.261 |
| teapot | 0.238 | 0.781 | 0.419 | 0.257 |
| cat2 | 0.261 | 0.773 | 0.572 | 0.255 |

**Comment by Rbrq03** [comment]
Yes, we also find this issue when trying to reproduce the number in other papers. Their values are also different from what I tested, and there seems to be no consistency between each article. Our conclusion is that the best way might be to re-evaluate all methods under your benchmark code to get a rank.
